runtime/onert/core/src/util/shapeinf/Onehot.cc

```diff
@@ -0,0 +1,119 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util/ShapeInference.h"
+
+namespace onert
+{
+namespace shape_inference
+{
+
+ir::Shape onehotShape(const ir::Shape &input_shape, const int depth, const int axis)
+{
+  int rank = input_shape.rank();
+
+  ir::Shape newShape(rank + 1); // output shape rank will be input rank + 1.
+
+  // ToDo : It will consider that case of the indice was ragged tensor.
+  if (rank == 1)
+  { // It means inputed indice is scalar.
+    int features = input_shape.num_elements();
+
+    if (axis == 0)
+    { // new_shape will be [batch] x [features].
+      newShape.dim(0) = depth;
+      newShape.dim(1) = features;
+    }
+    else
+    { // new_shape will be [features] x [depth].
+      newShape.dim(0) = features;
+      newShape.dim(1) = depth;
+    }
+  }
+  else if (rank == 2)
+  { // it means inputed indice is matrix(batch).
+    int batch = input_shape.dim(0);
+    int features = input_shape.dim(1);
+
+    if (axis == -1)
+    { // new_shape will be [batch] x [features] x [depth].
+      newShape.dim(0) = batch;
+      newShape.dim(1) = features;
+      newShape.dim(2) = depth;
+    }
+    else if (axis == 1)
+    { // new_shape will be [batch] x [depth] x [features].
+      newShape.dim(0) = batch;
+      newShape.dim(1) = depth;
+      newShape.dim(2) = features;
+    }
+    else
+    { // axis == 0
+      // new_shape will be [depth] x [batch] x [features].
+      newShape.dim(0) = depth;
+      newShape.dim(1) = batch;
+      newShape.dim(2) = features;
+    }
+  }
+
+  return newShape;
+}
+
+void StaticInferer::visit(const ir::operation::OneHot &op)
+{
+  const auto indice_idx{op.getInputs().at(ir::operation::OneHot::Input::INDICES)};
+  const auto &indice = _operands.at(indice_idx);
+
+  const auto depth = op.param().depth;
+  const auto axis = op.param().axis;
+
+  auto output_idx = op.getOutputs().at(0);
+  ir::Operand &output = _operands.at(output_idx);
+
+  if (!output.info().isDynamic())
+  {
+    output.info().setDynamic();
+  }
+
+  // re-sizing output shape
+  ir::Shape new_shape = onehotShape(indice.info().shape(), depth, axis);
+  output.info().shape(new_shape);
+}
+
+void DynamicInferer::visit(const ir::operation::OneHot &op)
+{
+  auto output_ind = op.getOutputs().at(0);
+  auto output = _tensor_registry->getITensor(output_ind);
+
+  auto indices_ind = op.getInputs().at(ir::operation::OneHot::INDICES);
+  const auto &indices = _tensor_registry->getITensor(indices_ind);
+  auto indices_shape = getShape(indices.get());
+
+  if (!output->is_dynamic())
```

Comment:
Could you check if this should be`(!input->is_dynamic)`? (please refer to other PRs).